src/test/java/com/iexec/common/docker/client/DockerClientInstanceTests.java

```diff
@@ -420,24 +424,24 @@ void shouldNotFindImagePresentSinceDockerCmdException() {
 
     @Test
     void shouldPullImage() {
+        dockerClientInstance.removeImage(ALPINE_LATEST);
         assertThat(dockerClientInstance.pullImage(ALPINE_LATEST)).isTrue();
         assertThat(dockerClientInstance.isImagePresent(ALPINE_LATEST)).isTrue();
-        dockerClientInstance.removeImage(ALPINE_LATEST);
     }
 
     @Test
     void shouldPullImageWithExplicitTimeout() {
+        dockerClientInstance.removeImage(ALPINE_LATEST);
         assertThat(dockerClientInstance.pullImage(ALPINE_LATEST, Duration.of(3, ChronoUnit.MINUTES))).isTrue();
         assertThat(dockerClientInstance.isImagePresent(ALPINE_LATEST)).isTrue();
-        dockerClientInstance.removeImage(ALPINE_LATEST);
     }
 
     @Test
     void shouldNotPullImageSinceTimeout() {
         dockerClientInstance.removeImage(ALPINE_LATEST);
         assertThat(dockerClientInstance.pullImage(ALPINE_LATEST, Duration.of(1, ChronoUnit.SECONDS))).isFalse();
         assertThat(dockerClientInstance.isImagePresent(ALPINE_LATEST)).isFalse();
-        dockerClientInstance.removeImage(ALPINE_LATEST);
+        dockerClientInstance.pullImage(ALPINE_LATEST);
```

Comment:
To ensure the image will be present before the next test.
Globally, there is a hidden assumption where this image needs to be present before each test method.
Here is the only test where the image is removed and not pulled again.
It could be done in a `beforeEach` but the tests would take a longer time to execute due to numerous unnecessary `docker pull` commands.

Reply:
Thank you for your answer.
"there is a hidden assumption where this image needs to be present before each test method"
There is a `pullImageIfNecessary()` method to use in order to avoid making hidden assumptions